File: Exemplos/03-Rect/rect.c
#include <stdio.h>

typedef struct {
  int x, y;
  int w, h;
} Rect;

void print (Rect* r) {
    printf (
        "Retangulo de tamanho (%d,%d) na posicao (%d,%d).\n",
        r->x,r->y, r->w,r->h
    );
}

void main (void) {
    Rect r1 = { 1,1, 10,10 };
    print(&r1);
}

File: Exemplos/09-Interface/figures/Rect.java
package figures;

import java.awt.*;

public class Rect extends Figure {
    public Rect (int x, int y, int w, int h) {
        super(x,y,w,h);
    }

    public void print () {
        System.out.format("Retangulo de tamanho (%d,%d) na posicao (%d,%d).\n",
            this.w, this.h, this.x, this.y);
    }

    public void paint (Graphics g) {
        Graphics2D g2d = (Graphics2D) g;
        g2d.drawRect(this.x,this.y, this.w,this.h);
    }
}
